conn.go

```diff
@@ -408,17 +397,47 @@ func (c *Conn) connect() error {
 			}
 		}
 
-		zkConn, err := c.dialer("tcp", c.Server(), c.connectTimeout)
+		dial := func() (net.Conn, error) {
+			ctx, cancel := context.WithTimeout(context.Background(), c.connectTimeout)
+			defer cancel()
+			return c.dialer.DialContext(ctx, "tcp", c.Server())
+		}
+
+		slog.Info("Dialing ZK server", "server", c.Server())
+		zkConn, err := dial()
 		if err == nil {
 			c.conn = zkConn
 			c.setState(StateConnected)
-			if c.logInfo {
-				c.logger.Printf("connected to %s", c.Server())
-			}
+			slog.Info("Connection established", "server", c.Server())
 			return nil
 		}
 
-		c.logger.Printf("failed to connect to %s: %v", c.Server(), err)
+		slog.Warn("Failed to connect to ZK server", "server", c.Server(), "err", err)
+		time.Sleep(addJitter(computeBackoff(attempt, c.reconnectInitialBackoff, c.reconnectMaxBackoff)))
```

Comment:
The  backoff should apply once all the servers in the DNS list have been tried. I remember that's how it used to work in the C client. Do you mind verifying the Java client's behavior?

Reply:
Can confirm the Java client only sleeps after trying all the hosts